src/main/scala/repo/Repository.scala

```diff
@@ -0,0 +1,23 @@
+package repo
```

Comment:
wrong package, interface of data access layer should be in domain because it is part of the domain

however, implementation like CustomerInMemoryRepository is part of infrastructure, so in infrastructure u provide implementation of CustomerRepository

why?

because domain is not interested in technical aspect of encoding / protocols / way of communcation (async vs sync) the place for this is infrastrcture